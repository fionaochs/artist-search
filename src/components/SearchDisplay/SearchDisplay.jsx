import React from 'react';
import PropTypes from 'prop-types';

const SearchDisplay = ({ onClick, onChange, name }) => (
  <section>
    <input type="text" value={name} name='artist name' placeholder="Search for artist" onChange={onChange}></input>
    <button onClick={onClick}>Search</button>
  </section>
);

SearchDisplay.propTypes = {
  onClick: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired
};

export default SearchDisplay;
