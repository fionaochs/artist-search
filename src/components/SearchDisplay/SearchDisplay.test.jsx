import React from 'react';
import { shallow } from 'enzyme';
import SearchDisplay from './SearchDisplay';

describe('SearchDisplay component', () => {
  it('renders SearchDisplay', () => {
    const wrapper = shallow(<SearchDisplay name='fefi' onChange={() => {}} onClick={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
