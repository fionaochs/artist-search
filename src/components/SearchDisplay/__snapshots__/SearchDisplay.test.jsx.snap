// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDisplay component renders SearchDisplay 1`] = `
<section>
  <input
    name="fefi"
    onChange={[Function]}
    placeholder="Search for artist"
    type="text"
    value="fefi"
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
</section>
`;
